### ISSUE TITLE
Improvement request Lora device bulk onboarding

### ISSUE THREAD

**Issue by phcrb** [SoftwareAG/cumulocity-lora#118]
Hi Cyril,
We're bulk onboarding sets of lora devices, that are described in an excel file

Data are formatted as shown:

| Name                | devEUI | appEUI | appKey | Type                                | Codec                          | Model   | Connectivity Plan     | Profile             |
|---------------------|--------|--------|--------|-------------------------------------|---------------------------------|---------|-----------------------|---------------------|
| SensorTH-VOC-xxxx    | aaaaa  | bbbbb  | ccccc  | gf_loradevice_enless_600-022        | Enless Wireless (Greenflex)     | 600-022 | orange-cs/CP_Basic     | Generic_classA_RX2SF12 |

When pasting the data into the onboarding form, only the fields Name, Dev EUI, App EUI and App Key are filled. Type and LNS related fields remain blank.

Can you improve this ?
Best regards
Philippe

**Comment by cpoder** [comment]
Hi,
I've added the "type" column to the copy/paste feature.
LNS related fields cannot be copied from an excel sheet however. The reason for that is that what you have in your excel sheet are labels, not ids.
It is though possible to copy and paste LNS related fields by provided them in the first line, then all added lines will have the sames values for those fields, whether you copy and paste values from an excel sheet or use the "+" button.
The only drawback is that you need to order your devices based on the LNS related fields if you want to copy and paste values efficiently.
So, basically, the steps to perform bulk provisioning with less copy and paste:
1. Group your devices in your Excel sheet per LNS related fields
2. In the first line of the bulk provisioning table, provide the values o the LNS related fields
3. In your Excel sheet, copy all fields from "name" to "type" which have the same LNS related field values
4. Paste them in the bulk provisioning table
5. If you have more devices in your Excel sheet with different LNS related fields, add a new line to the bulk provisioning table and restart from step 2

Best regards,
Cyril
